move/axelar_gateway/sources/gateway.move

```diff
@@ -291,11 +291,18 @@ public fun take_approved_message(
     );
 
     assert!(
-        self.messages[command_id].status == message.hash(),
+        is_message_approved(
+            self,
+            source_chain,
+            message_id,
+            source_address,
+            destination_id,
+            bytes32::from_bytes(hash::keccak256(&payload)),
+        ),
         EMessageNotApproved,
     );
 
-    self.messages[command_id].status = bytes32::new(MESSAGE_EXECUTED);
+    *table::borrow_mut(&mut self.messages, command_id) = MessageStatus::Executed;
```

Comment:
```suggestion
    *self.messages[command_id] = MessageStatus::Executed;
```
i think this should work. Separately, we should leverage the [index syntax](https://move-book.com/reference/index-syntax.html) for our own collections where possible

Reply:
it doesn't work, so i've tried index syntax. please take a look at [this commit](https://github.com/axelarnetwork/axelar-cgp-sui/pull/121/commits/9ac98fd5837cb58f5b690706569601819ef5ddd0)